plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultQueryBuilder.java

```diff
@@ -128,7 +129,9 @@ public PreparedQuery prepareJoinQuery(
         String query = format(
                 // The subquery aliases (`l` and `r`) are needed by some databases, but are not needed for expressions
                 // The joinConditions and output columns are aliased to use unique names.
-                "SELECT * FROM (SELECT %s FROM (%s) l) l %s (SELECT %s FROM (%s) r) r ON %s",
+                "SELECT %s, %s FROM (SELECT %s FROM (%s) l) l %s (SELECT %s FROM (%s) r) r ON %s",
+                formatProjectionAliases(client, leftProjections.values()),
+                formatProjectionAliases(client, rightProjections.values()),
```

Comment:
Can `leftProjections` and `rightProjections` intersect in absolute values?

Reply:
It was implemented this way in legacy.

Reply:
that was the intention
we could have an explicit check to ensure they never duplicate, and to convey the intention